src/main/java/org/alfresco/heartbeat/HBBaseDataCollector.java

```diff
@@ -36,12 +36,48 @@
   *
  * @author eknizat
  */
-public abstract class HBBaseDataCollector
-{
+public abstract class HBBaseDataCollector {
+    private String collectorId;
+    private String collectorVersion;
+    private String cronExpression;
+
+    /**
+     * The collector service managing this collector.
+     */
     private HBDataCollectorService hbDataCollectorService;
 
+    public HBBaseDataCollector(String collectorId)
```

Comment:
We need both, the collectorId and the version. Why do we pass one as constructor argument and the other one with a setter?

Reply:
I thought about this one as well, wanted to see if anyone will spot it.